src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,8 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+
+    //Making the method abstract forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.
+    public abstract boolean isMutating () ;
 }
```

Comment:
A more precise phrasing is `Returns true if …` (instead of `Checks if …`)
It is better to mention it returns true if the Command *potentially* mutates the data (header comments specify a contract, so try to be as precise as possible). E.g. Adding a duplicate person will not mutate data but the corresponding AddCommand object will still return true for this method.
@return … line can be omitted for this method as it says the same thing as the first sentence.